frontend/src/v5/ui/routes/dashboard/projects/federations/federationSettingsForm/federationSettingsForm.component.tsx

```diff
@@ -0,0 +1,405 @@
+/**
+ *  Copyright (C) 2022 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useEffect } from 'react';
+import { formatMessage } from '@/v5/services/intl';
+import { FormattedMessage } from 'react-intl';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as Yup from 'yup';
+import { TextField, MenuItem, InputLabel, FormControl } from '@material-ui/core';
+import { Select } from '@controls/select';
+import { FormModal } from '@/v5/ui/controls/modal/formModal/formDialog.component';
+import { useForm, SubmitHandler, Controller } from 'react-hook-form';
+import { useParams } from 'react-router';
+import { FederationsActionsDispatchers } from '@/v5/services/actionsDispatchers/federationsActions.dispatchers';
+import { IFederation, EMPTY_VIEW, RawFederationSettings } from '@/v5/store/federations/federations.types';
+import { ShareTextField } from '@controls/shareTextField';
+import { FlexContainer, SectionTitle, Thumbnail, ThumbnailPlaceholder, SelectView, ViewLabel, MenuItemView, UnitTextField } from './federationSettingsForm.styles';
+
+const UNITS = {
+	mm: 'Millimetres',
+	cm: 'Centimetres',
+	dm: 'Decimetres',
+	m: 'Metres',
+	ft: 'Feet and inches',
+};
+
+interface IFormInput {
+	name: string;
+	unit: string;
+	desc?: string;
+	code?: string;
+	defaultView: string;
+	latitude: number;
+	longitude: number;
+	angleFromNorth: number;
+	x: number;
+	y: number;
+	z: number;
+}
+
+const getDefaultValues = (federation: IFederation) => {
+	const { unit = 'mm', angleFromNorth = 0 } = federation.settings || {};
+	const {
+		latLong,
+		position,
+	} = federation.settings?.surveyPoint || {};
+	const [x, y, z] = position || [0, 0, 0];
+	const [latitude, longitude] = latLong || [0, 0];
+	const { code, name, description: desc } = federation;
+	const defaultView = federation?.settings?.defaultView || EMPTY_VIEW._id;
+	return {
+		name,
+		desc,
+		code,
+		unit,
+		defaultView,
+		latitude,
+		longitude,
+		angleFromNorth,
+		x,
+		y,
+		z,
+	};
+};
+
+const FederationSchema = Yup.object().shape({
+	name: Yup.string()
+		.min(2,
+			formatMessage({
```

Comment:
Move this out to a file in a new folder `src/v5/validation/`, and change it to FederationSettingsSchema.

Reply:
Done